### DISCUSSION TITLE
Tell me how to update data in a dynamic table.

### DISCUSSION THREAD

**Discussion by ZVektor** [ant-design-blazor/ant-design-blazor#4250] [Q&A]
Tell me how to update data in a dynamic table. When different List<Dictionary<string, object>> arrive. I have it planned that in a new method call, one of the keys is replaced with another, and the table seems to be waiting for the old key. Вместо ключа  'Дата выполнения операции'
![image](https://github.com/user-attachments/assets/ceecc7cc-4af4-48ae-a911-5ee8b216da90)
This is the old dictionary key, another one comes instead, but the table complains.

**Comment by ElderJames** [comment]
Thanks for contacting us @ZVektor , could you please post your code and I can take a look.

**Reply by ZVektor** [reply]
Hello, @ElderJames  I have published

**Comment by ZVektor** [comment]
Here is the table itself:
```csharp
<AntDesign.Table TItem="Dictionary<string, object>"
                 @ref="_table"
                 DataSource="@_dynamicData"
                 Loading="_isLoadingData"
                 EnableVirtualization
                 ScrollY="calc(100vh - 410px)"
                 Size="TableSize.Small"
                 Bordered
                 ScrollX="@((_dynamicData?.FirstOrDefault()?.Keys.Count * 150).ToString())"
                 HidePagination>
    @foreach (var key in _dynamicData?.FirstOrDefault()?.Keys.Take(10) ?? new string[0]) 
    {
        <AntDesign.PropertyColumn Property=@(c => c[key]) Title="@key" Width="150px"></AntDesign.PropertyColumn>
    }
</AntDesign.Table>
```
Variables are declared:
```csharp
List<Dictionary<string, object>> _dynamicData = new List<Dictionary<string, object>>();
private AntDesign.Table<Dictionary<string, object>> _table;
```

Click triggered method:
```csharp
protected async Task SendSqlQuery()
{
    try
    {
        _dynamicData = await carrigeService.ExecuteRawSqlQueryAsync(_sqlQuery, _timeoutSqlQueryInSeconds);
    }
    catch (Exception e)
    {
        ShowErrorNotification("Ошибка при загрузки данных", e.Message);
    }
    StateHasChanged();
}
```
Where _sqlQuery is a SQL query to the database.
This is the data I am receiving for the first time:
![image](https://github.com/user-attachments/assets/028af2a1-f67a-4c69-a6bc-f53feae7ddd1)
But the second time the table no longer wants to accept them.
![image](https://github.com/user-attachments/assets/619ce271-1493-4d42-9df5-c712b22c0640)

**Comment by ZVektor** [comment]
version AntDesign 0.20.2
